File: modules/spring-core/testdata/java/inspection/eventListener/src/com/event/listener/TestEventListener.java
package com.event.listener;

import org.springframework.context.event.EventListener;
import org.springframework.stereotype.Component;

@Component
public class TestEventListener {
	@EventListener
	void test0() {
	}

	@EventListener
	protected void test1() {
	}

	@EventListener
	private void test2() {
	}

	@EventListener
	public void test3() {
	}

	@EventListener
	public void test4(String param1, String param2) {
	}

	@EventListener
	private void test5(String param1, String param2) {
	}

	@EventListener
	public void test6(String param1) {
	}
}

File: modules/spring-core/testdata/kotlin/inspection/eventListener/src/com/event/listener/TestEventListener.kt
package com.event.listener

import org.springframework.context.event.EventListener
import org.springframework.stereotype.Component

@Component
class TestEventListener {
    @EventListener
    fun test0() {
    }

    @EventListener
    protected fun test1() {
    }

    @EventListener
    private fun test2() {
    }

    @EventListener
    fun test3() {
    }

    @EventListener
    fun test4(param1: String?, param2: String?) {
    }

    @EventListener
    private fun test5(param1: String, param2: String) {
    }

    @EventListener
    fun test6(param1: String?) {
    }
}